File: day3/second.py
import re
result = 0
with open("input.txt") as file:
    lines = file.read().splitlines()
    for i in range(0, len(lines), 3):
        group = lines[i:i+3]
        priority_char = [c for c in set.intersection(
            *(set(line) for line in group))][0]
        if re.match(r'^[a-z]$', priority_char):
            result += ord(priority_char) - 96  # 1 is the first value
        else:
            result += ord(priority_char) - 65 + 27
    print(result)

File: day3/second.rb
require 'set'
result = 0
File.open("input.txt") do |file|
  content = file.read()
  lines = content.split("\n")
  lines.each_slice(3) do |group|
    priority_char = group.map(&:chars).map(&:to_set).reduce(&:intersection).first
    if ('a'..'z').include?(priority_char)
      result += priority_char.ord - 96
    else
      result += priority_char.ord - 65 + 27
    end
  end
  puts result
end
